config/env/env.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2014 beego Author. All Rights Reserved.
+// Copyright 2017 Faissal Elamraoui. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package env
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+)
+
+var env struct {
+	data map[string]string
+	lock *sync.RWMutex
+}
+
+func init() {
+	env.data = make(map[string]string)
+	env.lock = &sync.RWMutex{}
+	for _, e := range os.Environ() {
+		splits := strings.Split(e, "=")
```

Comment:
I would like to check the splits's len here, it maybe cause panic if access the index 0